server/src/@types/User.d.ts

```diff
@@ -0,0 +1,43 @@
+//TODO do we need _id, timestamps passed in user data to FE?
+//TODO rewrite this and simplify, only need 1 or 2? 
+//TODO import { Document, Model } from "mongoose" -> extend the mongoose types with interfaces matching the user object
+
+declare namespace UserN {
+    interface UserLoginData {
+        password: string;
+        email: string;
+    }
+
+    interface UserRegistrationData extends UserLoginData {
+        firstName: string;
+        lastName: string;
+        isLoggedin: boolean;
+        username?: string;
```

Comment:
if username is optional, why have it? either none or obligatory I think

Reply:
for now I leave it the way it is, let's delete tomorrow